test/units/modules/test_known_hosts.py

```diff
@@ -14,11 +14,13 @@
 class KnownHostsDiffTestCase(unittest.TestCase):
 
     def _create_file(self, content):
-        tmp_file = tempfile.NamedTemporaryFile(prefix='ansible-test-', suffix='-known_hosts', delete=False)
-        tmp_file.write(to_bytes(content))
-        tmp_file.close()
-        self.addCleanup(os.unlink, tmp_file.name)
-        return tmp_file.name
+        tmp_filename = ''
+        with tempfile.NamedTemporaryFile(prefix='ansible-test-', suffix='-known_hosts', delete=False) as tmp_file:
+            tmp_file.write(to_bytes(content))
+            tmp_filename = tmp_file.name
+
+        self.addCleanup(os.unlink, tmp_filename)
```

Comment:
I think that the CM would delete the file prematurely in this specific instance. So the change is likely undesired here.